metabase/metabase
# Add anchor time setting to Admin > Caching

Part of #22538, based on the BE work from #22827

Adds an "Anchor time" setting input to Admin > Caching (`/admin/settings/caching`). People can control how often Metabase will be refreshing model caches. People can set the refresh interval (added in the previous PRs). We offer 1, 2, 3, 6, 12, and 24-hour options. Anchor time should make cache refresh helpful for longer refresh intervals (6 hours and more). For instance, you can set the anchor time to be 6am, a few hours before the workday begins. In that way, if your cache refresh interval is 6 hours and the last refresh has happened on, let's say, 11 pm, the first refresh the next day is going to happen at 6 am (the anchor time) as there's no need to refresh data at night and it's nice to start your day with some fresh model data :) 

### To Verify

1. Sign in as an admin
2. Go to `/admin/settings/caching`
3. Ensure you can't see the anchor time input when model persistence is disabled
4. Enable model persistence
5. Click through every "Refresh interval" option and make sure the anchor time field shows up for 6, 12, and 24-hour options
6. Make sure you can change the anchor time value, change between am/pm, etc.
7. Go to `/admin/settings/localization` and change the time format (12-hour vs. 24-hour)
8. Make sure the input adapts correctly

### Demo

https://user-images.githubusercontent.com/17258145/170296314-a8c88ed7-ab9b-4154-8a1b-53686d466c41.mp4
